cypress/support/commands.js

```diff
@@ -1,128 +1,325 @@
+                cy.get('[data-cy="add-allowed-origins"]').type('localhost:9000').should('have.value', 'localhost:9000');
+                cy.wait(2000);
+            });
+        }
+    })
+    cy.get('[data-cy="download-checkBox"]').check().should('be.checked');
+    cy.wait(2000);
+    cy.get('[data-cy="select-download-language"]').select('JavaScript');
+    cy.wait(2000);
+    cy.get('[data-cy="terms-condition-link"]').click();
+    cy.wait(2000);
+    cy.get('[data-cy="update-terms-condition"]').check().should('be.checked');
+    cy.wait(2000);
+    cy.get('[data-cy="create-credential-btn"]').click();
+    cy.wait(20000);
+    cy.get('button[data-cy="restart-download"]').click();
+    cy.wait(5000);
+    cy.checkCredential();
+    cy.wait(2000);
+    cy.get('[data-cy="Restart-new-credential"]').click();
+});
+
+Cypress.Commands.add(`checkPage`, () => {
+    cy.get('body').then(($body) => {
+        if ($body.find('[data-cy="credential-form"]').length > 0) {
+            cy.get('[data-cy="credential-form"]', { timeout: 10000 }).then(() => {
+                cy.addCredential();
+            });
+        }
+    });
+
+    cy.get('body').then(($body) => {
+        if ($body.find('[data-cy="return-flow"]').length > 0) {
+            cy.get('[data-cy="return-flow"]', { timeout: 10000 }).then(() => {
+                cy.checkReturnFlow();
+            });
+        }
+    });
+
+    cy.get('body').then(($body) => {
+        if ($body.find('[data-cy="request-access"]').length > 0) {
+            cy.get('[data-cy="request-access"]', { timeout: 10000 }).then(() => {
+                if ($body.find('[data-cy="accessDetails"]').length > 0) {
+                    cy.get('[data-cy="accessDetails"]', { timeout: 10000 }).then(() => {
+                        cy.checkRequestAccess();
+                    });
+                }
+                else if ($body.find('[data-cy="accessDetails-edgeCase"]').length > 0) {
+                    cy.get('[data-cy="accessDetails-edgeCase"]', { timeout: 10000 }).then(() => {
+                        cy.checkRequestAccessEdgeCase();
+                    });
+                }
+            });
+        }
+    });
+});
+
+Cypress.Commands.add(`enableOrganizationPicker`, () => {
+    cy.get('button[data-cy="change-organization-btn"]').click();
+    cy.wait(2000);
+    cy.get('button[data-cy="organization-picker"]').should('be.visible').should('be.enabled').click();
+});
+
+Cypress.Commands.add(`selectOrganization`, (elementAlias) => {
+    cy.get(elementAlias).click();
+    cy.wait(2000);
+    cy.get(elementAlias).then($el => {
+        const nextElement = $el.next().length > 0 ? $el.next() : false;
+        cy.get('button[data-cy="submit-change-organization"]').should('be.visible').should('be.enabled').click();
+        cy.wait(20000);
+        cy.checkPage();
+        cy.wait(3000)
+        if (nextElement) {
+            cy.enableOrganizationPicker();
+        }
+    });
+});

```

Comment:
@BaskarMitrah All these tests should be moved to `cypress/e2e/get-credentials.cy.js`

Reply:
Ok @dc297